### PR TITLE
correct documentation

### PR DESCRIPTION
**Pull request by ochriste**

### CODE REVIEW

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 1]
```diff
@@ -6,6 +6,9 @@ Version 1.6
 
 1. Get the new and heavily modified files:
 
+    .build/venv/bin/pcreate --interactive -s c2cgeoportal_create /tmp/{{project}} package={{package}} srid=-1
+    .build/venv/bin/pcreate --interactive -s c2cgeoportal_update /tmp/{{project}} package={{package}}
```
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Already done by c2ctool !

Comment (ochriste) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: it's absoluty unclear
if it's done by the c2ctool, you should say it before the cp commands

Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: unclear ? 
Do you follow this doc ?
http://docs.camptocamp.net/c2cgeoportal/1.6/integrator/upgrade_application.html#easy-upgrading-an-application-from-1-5-to-1-6

Comment (ochriste) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: I changed the PR

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 2]
```diff
@@ -6,6 +6,12 @@ Version 1.6
 
 1. Get the new and heavily modified files:
 
+    first follow the instruction as explained in the following documentation
+    chapter:
+    http://docs.camptocamp.net/c2cgeoportal/1.6/integrator/upgrade_application.html#easy-upgrading-an-application-from-1-5-to-1-6
```
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Shouldn't be put outside the step 1 ?

Comment (ochriste) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: I didnt want to renumber all the points, but if you think it's better, we
can do it

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 1 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: I just see it :-)

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 2 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Without any number for me it's general information not a task to do :-)

**Comment by sbrunner** [comment]
You also will have a correction in the CONST_Makfile ?

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 3]
```diff
@@ -4,7 +4,34 @@ This file includes migration steps for each release of c2cgeoportal.
 Version 1.6
 ===========
 
-1. Get the new and heavily modified files:
+1. Warning about .in and .mako
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: .in and .mako => ".in and .mako files" would be clearer

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 4]
```diff
@@ -4,7 +4,34 @@ This file includes migration steps for each release of c2cgeoportal.
...
+
+   In c2cgeoportal 1.6, the .in are globaly replaced by .mako
+
+   .in STILL WORK!
+
+   you can keep the .in, the system is backward compatible,
+   or you can choose to replace your .in by .mako, to stick with c2cgeoportal
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: stick to

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 5]
```diff
@@ -4,7 +4,34 @@ This file includes migration steps for each release of c2cgeoportal.
...
+   evolution.
+
+   .mako offer more flexibility, so it is adviced to replace the .in
+   by .mako
+
+   please note the following syntaxe change:
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: syntax

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 6]
```diff
@@ -4,7 +4,34 @@ This file includes migration steps for each release of c2cgeoportal.
...
+
+   a variable named ${vars:<varname>} in a .in will be written ${<varname>} in
+   a .mako
+
+   also, in all the CHANGELOG points specifing both .in/.mako files, it means
+   you need to modify the file you have (either the .in or the .mako), you MUST
+   NOT have both file.
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: files

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 7]
```diff
@@ -4,7 +4,34 @@ This file includes migration steps for each release of c2cgeoportal.
...
+
+2. Follow the instructions explained in this documentation chapter:
+    http://docs.camptocamp.net/c2cgeoportal/1.6/integrator/upgrade_application.html#easy-upgrading-an-application-from-1-5-to-1-6
+
+3. Get the new and heavily modified files:
+
+    at that point, the following folder should exists: /tmp/{{project}}/
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: exist

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 3 above)
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Thanks for this informative piece of text! Capitalizing the first letter of each sentence would be more correct.

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 8]
```diff
@@ -180,47 +210,51 @@ Version 1.6
         root =
             ${"\n    ".join(root)}
 
-15. Remove your Apache configuration file, probably in
+17. Remove your Apache configuration file, probably in
     `/var/www/vhost/{{project}}/conf`. Now it will be automatically created.
 
-16. Notes on database changes:
+    to define a customized path, set the variables APACHE_VHOST and/or
```
Comment (asaunier) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: to define => To define

File: doc/integrator/create_application.rst [hunk 9]
```diff
@@ -513,3 +513,54 @@ Then run the make command for the user/instance you want to setup:
...
+Several files are autogenerated, their content depending of the variables you
+have set either in the main ``<package>.mk`` or a ``<user>.mk``
+
+The files can have either the extension ``.in`` or ``.mako``
+
+.in
+...
+
+If you use ``.in``, the variables are simply replaced and a file without the
+``.in`` is generated.
```
Comment (asaunier) [inline doc/integrator/create_application.rst]: without the ``.in`` extension

File: doc/integrator/create_application.rst [hunk 10]
```diff
@@ -513,3 +513,54 @@ Then run the make command for the user/instance you want to setup:
...
+
+**Syntax**
+
+In ``.in`` files, the variable replacement syntax is as follow::
```
Comment (asaunier) [inline doc/integrator/create_application.rst]: as follows

File: doc/integrator/create_application.rst [hunk 11]
```diff
@@ -513,3 +513,54 @@ Then run the make command for the user/instance you want to setup:
...
+for example:
+
+* ``${vars:instanceid}``
+* ``${buildout:directory}``
+
+.mako
+.....
+
+If you use ``.mako``, you can also use all the possibilites allowed by the Mako
+templating system, allowing for loop, conditions, sub-templates, etc.
```
Comment (asaunier) [inline doc/integrator/create_application.rst]: allowing for loop, ... => such as for loops, ...

**Comment by ochriste** [comment]
@sbrunner 
> You also will have a correction in the CONST_Makfile ?
is there something I should do in that file ?

File: doc/integrator/create_application.rst [hunk 12]
```diff
@@ -513,3 +513,54 @@ Then run the make command for the user/instance you want to setup:
...
+``.in`` extension is generated.
+
+**Syntax**
+
+In ``.in`` files, the variable replacement syntax is as follows::
+
+  ${<configsectionname>:<variablename>}
+
+where the ``<configsectionname>`` is the name of section in the configuration
+where the variable is defined,
```
Comment (sbrunner) [inline doc/integrator/create_application.rst]: it was ..., now it's optional it can be specified only for backward compatibility ...

**Comment by sbrunner** [comment]
> is there something I should do in that file ?

as I see no ...

File: doc/integrator/create_application.rst (hunk 12 above)
Comment (ochriste) [inline doc/integrator/create_application.rst]: when was it modified? because in the version from 2 weeks ago it wasnt

Comment (sbrunner) [inline doc/integrator/create_application.rst]: In all version with make ..., I just test it and it's effectively working !

Comment (ochriste) [inline doc/integrator/create_application.rst]: mmm, im pretty sure I had that problem with the epfl project, so Im a bit doubtful

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 13]
```diff
@@ -251,6 +285,12 @@ Version 1.6
     git rm development.ini.in production.ini.in
     git rm versions.cfg
 
+26. In your `apache/wsgi.conf.in.mako` or `apache/wsgi.conf.in.in` file, do the
```
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: .in.mako , .in.in ?

**Comment by sbrunner** [comment]
Just a typo otherwise that looks good :-)

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 13 above)
Comment (ochriste) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: oups

**Comment by sbrunner** [comment]
Looks good, merge when Travis tests pass :-)

**Pull request merged**